### ISSUE TITLE
Hoverfly 1.5.0 seems not to forward form content in capture mode

### ISSUE THREAD

**Issue by csfulop** [SpectoLabs/hoverfly#1066]
## Description of the bug

Hoverfly 1.5.0 seems not to forward form content in capture mode

## Steps to reproduce the issue

Send POST request to http://validate.jsontest.com/ with json content as application/x-www-form-urlencoded.

Without hoverfly:
```
$ curl -v http://validate.jsontest.com/ -d 'json={"key":"value"}'
*   Trying 2a00:1450:400d:808::2013:80...
* TCP_NODELAY set
* Connected to validate.jsontest.com (2a00:1450:400d:808::2013) port 80 (#0)
> POST / HTTP/1.1
> Host: validate.jsontest.com
> User-Agent: curl/7.68.0
> Accept: */*
> Content-Length: 20
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 20 out of 20 bytes
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Access-Control-Allow-Origin: *
< Content-Type: application/json
< X-Cloud-Trace-Context: 35aecf7888482670c2a2bdb82491eb9d
< Date: Sat, 04 Mar 2023 15:57:17 GMT
< Server: Google Frontend
< Content-Length: 125
< 
{
   "size": 1,
   "parse_time_nanoseconds": 16547,
   "object_or_array": "object",
   "validate": true,
   "empty": false
}
* Connection #0 to host validate.jsontest.com left intact
```

With hoverfly 1.4.0:
```
$ ./hoverctl version

+----------+-------------+
| hoverctl | master-3828 |
| hoverfly | v1.4.0      |
+----------+-------------+
$ ./hoverctl start
Hoverfly is now running

+------------+------+
| admin-port | 8888 |
| proxy-port | 8500 |
+------------+------+
$ ./hoverctl mode capture
Hoverfly has been set to capture mode 
$ curl -v --proxy localhost:8500 http://validate.jsontest.com/ -d 'json={"key":"value"}'
*   Trying 127.0.0.1:8500...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8500 (#0)
> POST http://validate.jsontest.com/ HTTP/1.1
> Host: validate.jsontest.com
> User-Agent: curl/7.68.0
> Accept: */*
> Proxy-Connection: Keep-Alive
> Content-Length: 20
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 20 out of 20 bytes
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Access-Control-Allow-Origin: *
< Content-Length: 125
< Content-Type: application/json
< Date: Sat, 04 Mar 2023 15:56:19 GMT
< Hoverfly: Was-Here
< Server: Google Frontend
< X-Cloud-Trace-Context: d354ecd231d4f4e9f015e8ccc525d3a2
< 
{
   "size": 1,
   "parse_time_nanoseconds": 16146,
   "object_or_array": "object",
   "validate": true,
   "empty": false
}
* Connection #0 to host localhost left intact
```

With hoverfly 1.5.0:
```
$ ./hoverctl version

+----------+-------------+
| hoverctl | master-3950 |
| hoverfly | v1.5.0      |
+----------+-------------+
$ ./hoverctl start
Hoverfly is now running

+------------+------+
| admin-port | 8888 |
| proxy-port | 8500 |
+------------+------+
$ ./hoverctl mode capture
Hoverfly has been set to capture mode 
$ curl -v --proxy localhost:8500 http://validate.jsontest.com/ -d 'json={"key":"value"}'
*   Trying 127.0.0.1:8500...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8500 (#0)
> POST http://validate.jsontest.com/ HTTP/1.1
> Host: validate.jsontest.com
> User-Agent: curl/7.68.0
> Accept: */*
> Proxy-Connection: Keep-Alive
> Content-Length: 20
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 20 out of 20 bytes
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Access-Control-Allow-Origin: *
< Content-Length: 70
< Content-Type: application/json
< Date: Sat, 04 Mar 2023 15:58:24 GMT
< Hoverfly: Was-Here
< Server: Google Frontend
< X-Cloud-Trace-Context: 565ed2c31df44033f4340e60ebcea7bd
< 
{"error": "You must pass JSON via the ?json= parameter to validate."}
* Connection #0 to host localhost left intact
```

### Observed result

With hoverfly 1.5.0 http://validate.jsontest.com/ complaining about missing json data.

**Hoverfly error messages seen (If none, say none)**
```
none
```
**If possible, add screenshots to help explain your problem**

See the response of hoverfly and curl commands above.

### Expected result

http://validate.jsontest.com/ should receive the json data and answer that it is a valid json.

## Additional relevant information
1. Hoverfly version: 1.5.0
2. Anything that might help us to diagnose the problem
Without hoverfly and with previous 1.4.0 version of hoverfly the above scenario works.

**Comment by kapishmalik** [comment]
@csfulop we recently upgraded the Go version which has caused this. In the newer version of Go, if we do request.ParseForm then it automatically drains the request.Body. While reconstructing the request, we are sending an empty body. I am adding a fix for the same.

https://stackoverflow.com/questions/23773183/why-does-request-parseform-drain-the-request-body
https://groups.google.com/g/golang-nuts/c/mhiYjEymNDU

**Comment by tommysitu** [comment]
fixed in v1.5.1

**Comment by csfulop** [comment]
Thanks for the fix, I can confirm that it is OK with 1.5.1.
